### DISCUSSION TITLE
Simulate GPS

### DISCUSSION THREAD

**Discussion by alikxus** [bluenviron/gomavlib#35] [General]
Hi,  i need to Simulate gps over mavlink, but i do not understand how i can to send message using your library.

**Comment by aler9** [comment]
Hello, first you have to identify the message that is used to carry GPS informations. The mavlink protocol doesn't define a specific message for sending GPS informations, it just provides a list of messages that may or may not be used by a Drone.

You can either search in the Mavlink message list:
https://mavlink.io/en/messages/common.html

Or you can connect a gomavlib-based script to a Drone and listen for incoming messages:
https://github.com/aler9/gomavlib/blob/main/examples/endpoint-serial/main.go

If you don't have a Drone, you can use a simulator like Gazebo:
https://gazebosim.org

Once you've identified the message, you can send it with `node.WriteMessageAll()`.
